src/driver/libsql/LibSqlDriver.ts

```diff
@@ -0,0 +1,103 @@
+import type { createClient } from "@libsql/client"
+import { AbstractSqliteDriver } from "../sqlite-abstract/AbstractSqliteDriver"
+import type { LibSqlConnectionOptions } from "./LibSqlConnectionOptions"
+import { LibSqlQueryRunner } from "./LibSqlQueryRunner"
+import { PlatformTools } from "../../platform/PlatformTools"
+import type { QueryRunner } from "../../query-runner/QueryRunner"
+import type { DataSource } from "../../data-source/DataSource"
+import type { EntityMetadata } from "../../metadata/EntityMetadata"
+import { DriverPackageNotInstalledError } from "../../error/DriverPackageNotInstalledError"
+import type { ColumnType } from "../types/ColumnTypes"
+
+export class LibSqlDriver extends AbstractSqliteDriver {
+    sqlite: { createClient: typeof createClient }
+
+    options: LibSqlConnectionOptions
+
+    constructor(connection: DataSource) {
+        super(connection)
+        this.database = this.options.database
+
+        try {
+            this.sqlite = PlatformTools.load("libsql")
+        } catch (e) {
+            throw new DriverPackageNotInstalledError("SQLite", "sqlite3")
+        }
+    }
+
+    /**
+     * Closes connection with database.
+     */
+    async disconnect(): Promise<void> {
+        return new Promise<void>((ok, fail) => {
+            try {
+                this.queryRunner = undefined
+                this.databaseConnection?.close()
+                this.databaseConnection = undefined
+                ok()
+            } catch (error) {
+                fail(error)
+            }
+        })
+    }
+
+    /**
+     * Creates a query runner used to execute database queries.
+     */
+    createQueryRunner(): QueryRunner {
+        if (!this.queryRunner) this.queryRunner = new LibSqlQueryRunner(this)
+        return this.queryRunner
+    }
+
+    override createGeneratedMap(
+        metadata: EntityMetadata,
+        insertResult: any,
+        entityIndex: number,
+        entityNum: number,
+    ) {
+        if (typeof insertResult === "bigint")
+            insertResult = Number(insertResult)
+        return super.createGeneratedMap(
+            metadata,
+            insertResult,
+            entityIndex,
+            entityNum,
+        )
+    }
```

Comment:
This can be removed if https://github.com/tursodatabase/libsql-client-ts/issues/165 is addressed.